notification/notification.go

```diff
@@ -0,0 +1,93 @@
+package notification
```

Comment:
I'm not sure this belongs here. This is only used by the daemon to collect the notifications, right? The SDK only includes the code that test plans should rely on. I think this should go in the daemon.

The duplication of logic is also a bit of a smell.

Reply:
I think it belongs here, as ultimately we don't want the `daemon` to talk to Redis directly - we want it to pass by the `sync service`.

Then again you are right that this is to be used only by the daemon.

Reply:
But the daemon _is_ going to talk to Redis directly, right? (from the infra perspective). I think what you meant is that the intention was to encapsulate _everything_ related to Redis behind the sdk?

Reply:
> the intention was to encapsulate everything related to Redis behind the sdk?

precisely that.

the daemon is going to talk to Redis directly for now - this is a major change in this changeset!